hudi-utilities/src/test/java/org/apache/hudi/utilities/deltastreamer/TestTransformer.java

```diff
@@ -79,6 +96,71 @@ public void testMultipleTransformersWithIdentifiers() throws Exception {
     assertEquals(0, sqlContext.read().format("org.apache.hudi").load(tableBasePath).where("timestamp != 110").count());
   }
 
+  @Test
+  public void testTransformerSchemaValidation() throws Exception {
+    List<String> transformerClassNames = Arrays.asList(
+        FlatteningTransformerWithTransformedSchema.class.getName(),
+        TimestampTransformer.class.getName(),
+        AddColumnTransformerWithTransformedSchema.class.getName());
+    runDeltaStreamerWithTransformers(transformerClassNames);
+  }
+
+  @Test
+  public void testTransformerSchemaValidationFails() throws Exception {
```

Comment:
Should we break this into the different cases that are being tested so it is clear what failure mode is being covered?

Reply:
Addressed.